libs-scala/grpc-server-reflection-client/src/test/scala/com/daml/grpc/reflection/ServerReflectionClientSpec.scala

```diff
@@ -3,35 +3,28 @@
 
 package com.daml.grpc.reflection
 
-import java.util.concurrent.TimeUnit
-
+import com.daml.grpc.test.GrpcServer
+import io.grpc.StatusRuntimeException
 import io.grpc.health.v1.HealthGrpc
-import io.grpc.inprocess.{InProcessChannelBuilder, InProcessServerBuilder}
-import io.grpc.protobuf.services.ProtoReflectionService
 import io.grpc.reflection.v1alpha.ServerReflectionGrpc
-import io.grpc.services.HealthStatusManager
-import io.grpc.{BindableService, Channel, StatusRuntimeException}
-import org.scalatest.Assertion
-import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.flatspec.AsyncFlatSpec
 import org.scalatest.matchers.should.Matchers
 
-import scala.concurrent.Future
-
-final class ServerReflectionClientSpec extends AsyncFlatSpec with Matchers with ScalaFutures {
+final class ServerReflectionClientSpec extends AsyncFlatSpec with Matchers with GrpcServer {
 
   import ServerReflectionClientSpec._
+  import Services._
 
   behavior of "getAllServices"
 
   it should "fail if reflection is not supported" in withServices(health) { channel =>
     val stub = ServerReflectionGrpc.newStub(channel)
     val client = new ServerReflectionClient(stub)
-    client.getAllServices().failed.futureValue shouldBe a[StatusRuntimeException]
+    client.getAllServices().failed.map(_ shouldBe a[StatusRuntimeException])
```

Comment:
doesn’t this mean that now you’ll succeed if it doesn’t fail?

Reply:
It's the same as before, I just noticed that using `.futureValue` caused a deadlock in another test when using the serial `ExecutionContext` built-in into `AsyncFlatSpec`, so I preferred removing existing usages.

Reply:
So we suffered from the same issue before or it’s not an issue?

Reply:
I had this issue while creating the tests for the reverse proxy. I can revert it back to what it was before, it's just that I'm not fully aware of how or why `.futureValue` can deadlock and I would prefer to leave it out where possible.

Reply:
can we just turn this into an async test suite and wait on the future completing or if you don’t want to do that, `Await.result`? Looking through this again, it seems like the `map` will also executed asynchronously and it isn’t clear to me that the test cannot succeed before the future fails.

Reply:
> can we just turn this into an async test suite

Sorry, I lost you here, this is already an `AsyncFlatSpec`. 🤔 

> and wait on the future completing or if you don’t want to do that, `Await.result`? Looking through this again, it seems like the `map` will also executed asynchronously and it isn’t clear to me that the test cannot succeed before the future fails.

In an `AsyncFlatSpec` you are supposed to return a `Future[Assertion]` in your test blocks and Scalatest will take care of executing it correctly. You can test this by putting a failing assertion in the `map`.

Does this make sense or am I missing something?

Reply:
duh, I missed that this was already an async test suite and that this was the last statement. All my worries are unsubstantiated and this is perfectly fine. Sorry for the confusion